### ISSUE TITLE
Provide a specific error for a rule specification missing a plugin prefix

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [eslint/eslint#14027]
**The version of ESLint you are using.** 

7.18.0

**The problem you want to solve.**

In trying to enable a rule from a new custom plugin in my `.eslintrc.js`, I made the mistake of forgetting to include the plugin prefix in the specifier:

```js
{
    "extends": [
      "plugin:jest-react/recommended"
    ],
    "plugins": [
      "jest-react"
    ],
    "rules": {
        // This should have been:
        // "jest-react/no-mocking-react": "error"
        "no-mocking-react": "error"
    }
  }
```

**Your take on the correct solution to problem.**

It would be nice if a _"did you mean?"_ could be printed at the end of all outputs for each rule that could be matched but for a plugin prefix. Roughly proposing:

```text
/path/to/file1.js
  1:1  error  Definition for rule 'no-mocking-react' was not found (though a jest/  no-mocking-react

Definition for 'no-mocking-react' was not found, but 'jest-react/no-mocking-react' was.
```

**Are you willing to submit a pull request to implement this change?**

It would be my honor and privilege. ❤️ 

I dug a little bit and saw that rules are stored in `slots.lastConfigArray.pluginRules` within `Linter`. If a rule is not found, we could check if (semi-pseudocoding here)...

```js
Array.from(slots.lastConfigArray.pluginRules.entries()).some(ruleEntryKey => ruleEntryKey === `/${id}`)
```

**Comment by anikethsaha** [comment]
Thanks for the issue, 

This is an interesting suggestion and definitely a helpful one. 
My only concern is that it might take some more time to look up and search all plugins when there are too many plugins. 
Though this will not be a big concern still it would be good to handle if possible.

+1 for this

**Comment by nzakas** [comment]
At the moment, we aren't making any further changes to how config works or is used while we are building out the [new config system](https://github.com/eslint/eslint/issues/13481).

I wouldn't mind looking at this again once the new config system is up and running, as I think this is a nice addition to help users out.

**Comment by nzakas** [comment]
Created a PR for this in flat config :)
